autonomi/src/client/data_types/pointer.rs

```diff
@@ -71,60 +109,167 @@ impl Client {
+                    RecordKind::DataWithPayment(DataTypes::Pointer),
+                )
+                .map_err(|_| PointerError::Serialization)?
+                .to_vec(),
+                publisher: None,
+                expires: None,
+            };
+            (record, payees)
+        } else {
+            let record = Record {
+                key: NetworkAddress::from_pointer_address(address).to_record_key(),
+                value: try_serialize_record(&pointer, RecordKind::DataOnly(DataTypes::Pointer))
+                    .map_err(|_| PointerError::Serialization)?
+                    .to_vec(),
+                publisher: None,
+                expires: None,
+            };
+            (record, None)
+        };
+
+        let get_cfg = GetRecordCfg {
+            get_quorum: Quorum::Majority,
+            retry_strategy: Some(RetryStrategy::default()),
+            target_record: None,
+            expected_holders: Default::default(),
+        };
+
+        let put_cfg = PutRecordCfg {
+            put_quorum: Quorum::All,
+            retry_strategy: None,
+            verification: Some((VerificationKind::Crdt, get_cfg)),
+            use_put_record_to: payees,
+        };
+
+        // store the pointer on the network
+        debug!("Storing pointer at address {address:?} to the network");
+        self.network
+            .put_record(record, &put_cfg)
+            .await
+            .inspect_err(|err| {
+                error!("Failed to put record - pointer {address:?} to the network: {err}")
+            })?;
+
+        Ok((total_cost, address))
+    }
+
+    /// Create a new pointer on the network
+    /// Make sure that the owner key is not already used for another pointer as each key is associated with one pointer
+    pub async fn pointer_create(
+        &self,
+        owner: &SecretKey,
+        target: PointerTarget,
+        wallet: &EvmWallet,
+    ) -> Result<(AttoTokens, PointerAddress), PointerError> {
+        let address = PointerAddress::from_owner(owner.public_key());
+        let already_exists = match self.pointer_get(address).await {
+            Ok(_) => true,
+            Err(PointerError::Network(NetworkError::GetRecordError(
+                GetRecordError::SplitRecord { .. },
+            ))) => true,
+            Err(PointerError::Network(NetworkError::GetRecordError(
+                GetRecordError::RecordNotFound,
+            ))) => false,
+            Err(err) => return Err(err),
+        };
+
+        if already_exists {
+            return Err(PointerError::PointerAlreadyExists(address));
+        }
+
+        let pointer = Pointer::new(owner, 0, target);
+        self.pointer_put(pointer, wallet).await
+    }
+
+    /// Update an existing pointer to point to a new target on the network
+    /// The pointer needs to be created first with [`Client::pointer_put`]
+    /// This operation is free as the pointer was already paid for at creation
+    pub async fn pointer_update(
```

Comment:
there are now `pointer_create`, `pointer_update` and `point_put` pub functions
it may only need  `pointer_create` and `pointer_update` functions to be public, and `point_put` being a private helper function to be used by both others ?

Reply:
Maybe, wondering if some folk want to go the manual way and do more creative stuff with pointers?